### PR TITLE
add extension types, remove path types

### LINKED ISSUE #23: Extension type

### PR DESCRIPTION
**Pull request by jonsterling**
To resolve #23

### CODE REVIEW

File: src/lib/Monads.ml [hunk 1]
```diff
@@ -144,6 +144,10 @@ struct
     M.scope @@ fun local ->
     {local with size = i + local.size}
 
+  let globally m =
+    m |> M.scope  @@ fun local ->
```
Comment (favonia) [inline src/lib/Monads.ml:148]: Mortal space characters detected.

**Comment by jonsterling** [comment]
@favonia So there is a slight snag lol (which may not matter): the global scope of the family of cofibrations in the syntax means that we can't actually write down a `TermBuilder` for `ext` in a useful way --- that is, you cannot do something like `Splice.con phi @@ Splice.term @@ TB.ext n phi ...`, because `phi` is a variable and there are of course no variables allowed in that position. With a notion of "metavariable" we could overcome this, but it may be overkill.

Something to watch out for.

**Comment by jonsterling** [comment]
(This would also be solved by a full-fledged necessity modality!)

**Comment by favonia** [comment]
> @favonia So there is a slight snag lol (which may not matter): the global scope of the family of cofibrations in the syntax means that we can't actually write down a `TermBuilder` for `ext` in a useful way --- that is, you cannot do something like `Splice.con phi @@ Splice.term @@ TB.ext n phi ...`, because `phi` is a variable and there are of course no variables allowed in that position. With a notion of "metavariable" we could overcome this, but it may be overkill.
> 
> Something to watch out for.

Did you mean `Splice.con phi @@ fun phi ->`?

**Comment by jonsterling** [comment]
yes

**Comment by jonsterling** [comment]
I guess all this can also be resolved (if needed) by secretly adding a "validity context" in the style of Pfenning & Davies.

File: src/lib/TermBuilder.ml [hunk 2]
```diff
@@ -225,7 +225,7 @@ let code_sg mbase mfam : _ m =
 let code_path mfam mbdry : _ m =
   let+ fam = mfam
   and+ bdry = mbdry in
-  S.CodePath (fam, bdry)
+  S.CodeExt (1, fam, `Global (S.Lam (`Anon, S.Cof (Cof.Join [S.Cof (Cof.Eq (S.Var 0, S.Dim0)); S.Cof (Cof.Eq (S.Var 0, S.Dim1))]))), bdry)
```
Comment (favonia) [inline src/lib/TermBuilder.ml:228]: @jonsterling could this be refactored with cof_to_con and Cof.boundary?

Comment (jonsterling) [reply src/lib/TermBuilder.ml:228]: @favonia Note that these are all syntax, not domain. So no.

Comment (favonia) [reply src/lib/TermBuilder.ml:228]: :sob:

**Pull request merged**